cmd/go-sbot/config.go

```diff
@@ -0,0 +1,175 @@
+	err = decoder.Decode(&conf.presence)
+	check(err, "decode into presence map")
+
+	return conf
+}
+
+func ReadEnvironmentVariables(config *SbotConfig) {
+	if val := os.Getenv("SSB_SECRET_FILE"); val != "" {
+		loglib.Fatalln("flag SSB_SECRET_FILE not implemented")
+	}
+	if val := os.Getenv("SSB_SOCKET_FILE"); val != "" {
+		loglib.Fatalln("flag SSB_SOCKET_FILE not implemented")
+	}
+	if val := os.Getenv("SSB_LOG_LEVEL"); val != "" {
+		loglib.Fatalln("flag SSB_LOG_LEVEL not implemented")
+	}
+	if val := os.Getenv("SSB_CAP_INVITE_KEY"); val != "" {
+		loglib.Fatalln("flag SSB_CAP_INVITE_KEY not implemented")
+	}
+
+	// go-ssb specific env flag, for peachcloud/pub compat
+	if val := os.Getenv("GO_SSB_REPAIR_FS"); val != "" {
+		config.RepairFSBeforeStart = BooleanIsTrue(val)
+		config.presence["repair"] = true
+	}
+
+	if val := os.Getenv("SSB_DATA_DIR"); val != "" {
+		config.Repo = val
+		config.presence["repo"] = true
+	}
+
+	// handled in /cmd/go-sbot
+	// if val := os.Getenv("SSB_CONFIG_FILE"); val != "" {
+	// }
+
+	if val := os.Getenv("SSB_LOG_DIR"); val != "" {
+		config.DebugDir = val
+		config.presence["debugdir"] = true
+	}
+	if val := os.Getenv("SSB_PROMETHEUS_ADDRESS"); val != "" {
+		config.MetricsAddress = val
+		config.presence["debuglis"] = true
+	}
+	if val := os.Getenv("SSB_PROMETHEUS_ENABLED"); val != "" {
+		config.presence["debuglis"] = BooleanIsTrue(val)
+	}
+
+	if val := os.Getenv("SSB_HOPS"); val != "" {
+		hops, err := strconv.Atoi(val)
+		check(err, "parse hops from environment variable")
+		config.Hops = uint(hops)
+		config.presence["hops"] = true
+	}
+	if val := os.Getenv("SSB_MUXRPC_ADDRESS"); val != "" {
+		config.MuxRPCAddress = val
+		config.presence["lis"] = true
+	}
+	if val := os.Getenv("SSB_WS_ADDRESS"); val != "" {
+		config.WebsocketAddress = val
+		config.presence["wslis"] = true
+	}
+	if val := os.Getenv("SSB_EBT_ENABLED"); val != "" {
+		config.EnableEBT = BooleanIsTrue(val)
+		config.presence["enable-ebt"] = true
+	}
+	if val := os.Getenv("SSB_CONN_FIREWALL_ENABLED"); val != "" {
+		config.EnableFirewall = BooleanIsTrue(val)
+		config.presence["promisc"] = true
+	}
+	if val := os.Getenv("SSB_SOCKET_ENABLED"); val != "" {
+		config.NoUnixSocket = !BooleanIsTrue(val)
+		config.presence["nounixsock"] = true
+	}
+	if val := os.Getenv("SSB_CONN_DISCOVERY_UDP_ENABLED"); val != "" {
+		config.EnableDiscoveryUDP = BooleanIsTrue(val)
+		config.presence["localdiscov"] = true
+	}
+	if val := os.Getenv("SSB_CONN_BROADCAST_UDP_ENABLED"); val != "" {
+		config.EnableAdvertiseUDP = BooleanIsTrue(val)
+		config.presence["localadv"] = true
+	}
+	if val := os.Getenv("SSB_CAP_SHS_KEY"); val != "" {
+		config.ShsCap = val
+		config.presence["shscap"] = true
+	}
+	if val := os.Getenv("SSB_CAP_HMAC_KEY"); val != "" {
+		config.Hmac = val
+		config.presence["hmac"] = true
+	}
+}
+
+func BooleanIsTrue(s string) bool {
+	return s == "true" || s == "1" || s == "yes" || s == "on"
+}
```

Comment:
Suggest using a custom json.Unmarshaler ala

```go

type ConfigBool bool

func (cfg ConfigBool) UnmarshalJSON(b []byte) error {
	var raw string
	err := json.Unmarshal(b, &raw)
	if err != nil {
		return err
	}


        *cfg = s == "true" || s == "1" || s == "yes" || s == "on"

        if !cfg {
               // TODO: check if s == "false"  || s == "0" etc to rais an error
        }

        return nil
}
```

Reply:
converting into draft while i get my bools in order :neckbeard:

Reply:
got it 💪